src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Documents/VisualStudioEditorDocumentManager.cs

```diff
@@ -219,10 +221,13 @@ public void DocumentRenamed(uint cookie, string fromFilePath, string toFilePath)
         }
     }
 
+    [MemberNotNull(nameof(_runningDocumentTable))]
     private void EnsureDocumentTableAdvised()
     {
         JoinableTaskContext.AssertUIThread();
 
+        _runningDocumentTable ??= _serviceProvider.GetService<SVsRunningDocumentTable, IVsRunningDocumentTable4>(throwOnFailure: true).AssumeNotNull();
```

Comment:
Probably worth adding a note why this is late created and how it requires the UI thread for the `GetService` call

Reply:
Thanks! I'll add a note. Although this is kind of business as usual for VS, isn't it?

Reply:
Yes, but I think the more we remind people the less likely it is for someone to break it